core/arch/arm/kernel/trace_ext.c

```diff
@@ -28,20 +28,39 @@
 #include <trace.h>
 #include <console.h>
 #include <kernel/thread.h>
+#include <kernel/spinlock.h>
+#include <mm/core_mmu.h>
 
 const char trace_ext_prefix[] = "TEE-CORE";
 int trace_level = TRACE_LEVEL;
+static unsigned int puts_lock = SPINLOCK_UNLOCK;
 
 void trace_ext_puts(const char *str)
 {
+	uint32_t itr_status = thread_mask_exceptions(THREAD_EXCP_FOREIGN_INTR);
```

Comment:
The idea that introducing mask/unmask is that cpu_spin_lock/unlock use assert to check irq. I do not think this is good to use assert here, but no alternative from me, so just keep it. Because smp cores will output info to uart, we do not want a mess. The best solution is to use cpu_spin_lock_irqsave/restore, but not implemented now. So use mask/unmask exception here. I do not think masking all exceptions is a good idea.